sdk/synapse/azure-analytics-synapse-accesscontrol/src/main/java/com/azure/analytics/synapse/accesscontrol/AccessControlAsyncClient.java

```diff
@@ -0,0 +1,497 @@
+     * @return role by role Id.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SimpleResponse<SynapseRole>> getRoleDefinitionByIdWithResponse(String roleId, Context context) {
+        return this.serviceClient.getRoleDefinitionByIdWithResponseAsync(roleId, context);
+    }
+
+    /**
+     * Get role by role Id.
+     * 
+     * @param roleId Synapse Built-In Role Id.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return role by role Id.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SynapseRole> getRoleDefinitionById(String roleId) {
+        return this.serviceClient.getRoleDefinitionByIdAsync(roleId);
+    }
+
+    /**
+     * Get role by role Id.
+     * 
+     * @param roleId Synapse Built-In Role Id.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return role by role Id.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SynapseRole> getRoleDefinitionById(String roleId, Context context) {
+        return this.serviceClient.getRoleDefinitionByIdAsync(roleId, context);
+    }
+
+    /**
+     * Get the next page of items.
+     * 
+     * @param nextLink null
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return a list of Synapse roles available.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public Mono<PagedResponse<SynapseRole>> getRoleDefinitionsNextSinglePage(String nextLink) {
+        return this.serviceClient.getRoleDefinitionsNextSinglePageAsync(nextLink);
+    }
+
+    /**
+     * Get the next page of items.
+     * 
+     * @param nextLink null
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return a list of Synapse roles available.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public Mono<PagedResponse<SynapseRole>> getRoleDefinitionsNextSinglePage(String nextLink, Context context) {
```

Comment:
This doesn't follow guidelines on how to provide paginated responses to the user.

Reply:
Could you please explain a little bit more? What should it look like?